api/all/src/main/java/io/opentelemetry/api/GlobalOpenTelemetry.java

```diff
@@ -67,10 +83,68 @@ public final class GlobalOpenTelemetry {
   private GlobalOpenTelemetry() {}
 
   /**
-   * Returns the registered global {@link OpenTelemetry}.
+   * Returns the registered global {@link OpenTelemetry} if set, or else {@link
+   * OpenTelemetry#noop()}.
+   *
+   * <p>NOTE: if the global instance is set, the response is obfuscated to prevent callers from
+   * casting to SDK implementation instances and inappropriately accessing non-instrumentation APIs.
+   *
+   * <p>NOTE: This does not result in the {@link #set(OpenTelemetry)} side effects of {@link
+   * #get()}.
+   *
+   * <p>Native instrumentation should use this method to initialize their default {@link
+   * OpenTelemetry} instance. See class javadoc for more details.
+   */
+  public static OpenTelemetry getOrNoop() {
+    synchronized (mutex) {
+      return globalOpenTelemetry != null ? globalOpenTelemetry : OpenTelemetry.noop();
+    }
+  }
+
+  /**
+   * Returns the registered global {@link OpenTelemetry} if set, or else calls {@link
+   * #set(Supplier)} with the {@code supplier}.
+   *
+   * <p>NOTE: if the global instance is set, the response is obfuscated to prevent callers from
+   * casting to SDK implementation instances and inappropriately accessing non-instrumentation APIs.
+   *
+   * <p>NOTE: This does not result in the {@link #set(OpenTelemetry)} side effects of {@link
+   * #get()}.
+   *
+   * <p>Application custom instrumentation should use this method during initialization. See class
+   * javadoc for more details.
+   */
+  public static OpenTelemetry getOrSet(Supplier<OpenTelemetry> supplier) {
+    synchronized (mutex) {
+      if (globalOpenTelemetry == null) {
+        OpenTelemetry openTelemetry = supplier.get();
+        set(openTelemetry);
+        return openTelemetry;
```

Comment:
Not obfuscated here

Reply:
This was intentional, but now I'm having second thoughts. I was thinking an application owner calling this with a `Supplier` invoked would want to be able to access the `OpenTelemetrySdk` instance returned by the supplier so they could manage its lifecycle.

But maybe its better to always obfuscate, and tell callers that the `Supplier` needs to keep a ref to `OpenTelemetrySdk` to handle lifecycle stuff. 

Also not sure how to reconcile this with `getOrNoop()` returning unobfuscated noop instance. This is nice because callers can check if `getOrNoop() == OpenTelemetry.noop()`, but the inconsistency is a code smell. When and why do we return obfuscated vs. unobfuscated instances?

Reply:
Noop should never be obfuscated IMO